components/merkledb/src/group.rs

```diff
@@ -106,6 +108,69 @@ where
     }
 }
 
+impl<T, K, I> Group<T, K, I>
+where
+    T: RawAccess,
+    K: BinaryKey + Clone, //TODO: This clone definitely should be removed.
+    I: FromAccess<T>,
+{
+    ///TODO: add doc
+    pub fn iter<V: BinaryValue>(&self) -> GroupIter<K, T, I> {
+        let prefix = IndexAddress::from(self.prefix.clone());
+        let indexes_pool = IndexesPoolWrapper::new(self.access.clone());
+        let keys = indexes_pool.suffixes(&prefix);
+
+        GroupIter::new(self.prefix.clone(), keys, self.access.clone())
+    }
+}
+
+///TODO: add doc
+#[derive(Debug)]
+pub struct GroupIter<K, T, I> {
+    base: IndexAddress,
+    keys: Vec<K>,
```

Comment:
Storing all keys in the memory looks *wildly* inefficient in terms of memory use. OTOH, building right on top of a `IndexesPool` iterator is impossible - since `IndexesPool` would be borrowed, this would mean that operation on any other index is impossible during iteration! Honestly, I don't know what's the best solution here; using a random access on `IndexesPool` once the iterator is advanced seems viable, but is probably inefficient (in terms of performance). It could make sense to simply have a method in `Group` returning `Vec<K>` (a vector of all keys in the group). This would clearly notify the user that the method can use a large amount of RAM.

Reply:
But what if we change `iter` signature from `&self` to `self` to take onwership of the `IndexesPool`?

Reply:
What will this change? For example, if the `IndexesPool` is built on top of a `&Fork`, it can be instantiated at most once at any given time. If we have one instance in the iterator, we cannot have another instance anywhere else, i.e., we cannot instantiate new indexes during iteration. (I suppose with some tinkering it may become possible to instantiate indexes we're iterating over, but it's still not enough IMO.)